view/ce_dashboard/flask/blueprints/overview.py

```diff
@@ -1,28 +1,49 @@
 from flask import Blueprint, render_template, request
 import csv
-import urllib
-import urllib.parse
 import urllib.request
 import functools
 import xml.etree.ElementTree as ET
 from utils import cache_response_to_disk, make_data_response
+import time
 
 #######################
-# Functions to query Topology and cache responses in RAM
+# Functions to query Topology and cache responses
 #######################
 
-@functools.lru_cache(maxsize=1)
-def _get_projects_tree() -> ET.Element:
+# Query the Topology database for project info, storing the result in a local file for 1 hour
+# We store it in a local file so that we can share the response between multiple wsgi server
+# processes, and to avoid hitting the topology for each process.
+@cache_response_to_disk(file_name="topology_projects.bin", seconds_to_cache=60*60)
+def _get_topology_projects_xml():
```

Comment:
I don't see any exception handling in either this method (or further up in its call chain).  If a topology download fails, the exception will bubble up and crash the program (or at least result in a 503) which is probably not what you want.